sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -87,25 +104,19 @@ def _generate_recommendations(self):
         # Always recommend all products already present in the linked SO
         for line in self.order_id.order_line:
             found_line = found_dict.get(line.product_id.id, {})
-            new_line = recommendation_lines.new({
-                "product_id": line.product_id.id,
-                "times_delivered": found_line.get("product_id_count", 0),
-                "units_delivered": found_line.get("qty_delivered", 0),
-                "units_included": line.product_uom_qty,
-                "sale_line_id": line.id,
-            })
+            new_line = recommendation_lines.new(
+                self._prepare_recommendation_line_vals(found_line, line)
```

Comment:
this only will work if the product is already in the `found_dict`. Maybe it's a matter of just getting it in the `_prepare_recommendation_line_vals`

Reply:
Fixed!